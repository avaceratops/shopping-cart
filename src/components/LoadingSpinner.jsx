import ClipLoader from 'react-spinners/ClipLoader';

export default function LoadingSpinner() {
  return (
    <div className="grid h-72 place-content-center">
      <ClipLoader
        color={'#6366f1'}
        loading={true}
        cssOverride={{}}
        size={80}
        aria-label="Loading Spinner"
        data-testid="loader"
      />
    </div>
  );
}
