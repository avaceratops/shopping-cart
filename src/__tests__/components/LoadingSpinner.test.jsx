import { render } from '@testing-library/react';
import LoadingSpinner from '../../components/LoadingSpinner';

it('renders loading spinner correctly', () => {
  const { container } = render(<LoadingSpinner />);
  expect(container).toMatchSnapshot();
});
