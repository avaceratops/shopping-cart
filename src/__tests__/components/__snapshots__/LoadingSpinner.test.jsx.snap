// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`renders loading spinner correctly 1`] = `
<div>
  <div
    class="grid h-72 place-content-center"
  >
    <span
      aria-label="Loading Spinner"
      data-testid="loader"
      style="width: 80px; height: 80px; border-radius: 100%; border: 2px solid; border-top-color: #6366f1; border-bottom-color: transparent; border-left-color: #6366f1; border-right-color: #6366f1; display: inline-block; animation: react-spinners-ClipLoader-clip 0.75s 0s infinite linear; animation-fill-mode: both;"
    />
  </div>
</div>
`;
